From 18f7f64e3cc86f0a66c6a1524f63168142c2392f Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Wed, 18 Oct 2017 19:39:45 +0100
Subject: [PATCH 3/4] StatusBarSettings: Inject into Settings via IA

 * Make use of the new information architecture for the Settings app
   and place the status bar settings into System category, right bellow
   "System profiles" entry by setting a lower intent priority.

Change-Id: Ic36716b87c5ff61ac031ca87e0874bedc7a53c02
---
 AndroidManifest.xml | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 6106148..c34aaf2 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -171,31 +171,24 @@
                 android:resource="@string/summary_empty" />
         </activity-alias>
 
-        <!-- Status bar settings (dashboard) -->
-        <!--
+        <!-- Status bar settings (System category) -->
         <activity-alias
             android:name=".statusbar.StatusBarSettings"
             android:label="@string/status_bar_title"
+            android:icon="@drawable/ic_settings_statusbar"
             android:targetActivity="PartsActivity">
-            <intent-filter android:priority="2">
-                <action android:name="com.android.settings.action.EXTRA_SETTINGS" />
+            <intent-filter android:priority="254">
+                <action android:name="org.lineageos.lineageparts.action.SETTINGS" />
                 <action android:name="org.lineageos.lineageparts.STATUS_BAR_SETTINGS" />
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
             <meta-data
                 android:name="com.android.settings.category"
-                android:value="com.android.settings.category.personal" />
-            <meta-data
-                android:name="com.android.settings.icon"
-                android:resource="@drawable/ic_settings_statusbar" />
-            <meta-data
-                android:name="org.lineageos.settings.summary.receiver"
-                android:value="org.lineageos.lineageparts.PartsUpdater" />
+                android:value="com.android.settings.category.ia.system" />
             <meta-data
-                android:name="org.lineageos.settings.summary.key"
-                android:value="status_bar_settings" />
+                android:name="com.android.settings.summary"
+                android:resource="@string/summary_empty" />
         </activity-alias>
-        -->
 
         <!-- Anonymous Statistics -->
         <!--
-- 
2.7.4

